src/Core/Domain/Alias/Query/SearchAliasesForAssociation.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Alias\Query;
+
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\AliasConstraintException;
+
+/**
+ * Class SearchAliasesForAssociation is responsible for searching aliases with particular search terms.
+ */
+class SearchAliasesForAssociation
+{
+    public function __construct(public readonly string $searchTerm, public readonly ?int $limit = null)
+    {
+        if (null !== $limit && $limit <= 0) {
+            throw new AliasConstraintException('Search limit must be a positive integer or null', AliasConstraintException::INVALID_SEARCH_LIMIT);
```

Comment:
im not sure this kind of input needs a validation. Doesn't bring much value, especially because it doesn't write to db, so I think null can be treated same as 0 and negative values can be casted to positive (you can search for examples of such queries in product domain, I think we never overcomplicated these things)...

** EDIT**
oh ok there is at least this `SearchProductsForAssociation` which asserts the value as well. Though I think we shouldn't create dedicated domain exception for each of those cases, it doesn't seem to be worth it. There could be one exception then (as the limit is actually more of an infrastructural stuff, rather than domain :man_shrugging: ) Maybe even InvalidArgumentException from core suits it better? wdyt @jolelievre?

Reply:
I think checking that limit is not negative makes sense, but it's not a constraint for the domain indeed It's better to use `InvalidArgumentException` in this case indeed